OptimizelySDK/Entity/FeatureDecision.cs

```diff
@@ -0,0 +1,52 @@
+﻿/* 
+ * Copyright 2017, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace OptimizelySDK.Entity
+{
+    public class FeatureDecision : IEquatable<object>
+    {
+        public const string DECISION_SOURCE_EXPERIMENT = "experiment";
+        public const string DECISION_SOURCE_ROLLOUT = "rollout";
+
+        public string ExperimentId { get; set; }
+        public string VariationId { get; set; }
+        public string Source { get; set; }
+
+        public FeatureDecision(string experimentId, string variationId, string source)
```

Comment:
looks extra code to me, if properties are public then use 
new {prop=1, prop=2};

Reply:
LOL ... Looks like a good example of conflict of opinions here :)

For me, creating FeatureDecision instance using FeatureFlag {ExperimentId = expId, VariationId = varId, Source = source} again and again adds lot more code then constructor i.e. FeatureDecision( expId, varId, source).
Object initializer syntax looks more suitable for situations where some properties are optional which is not the case here.

Also, we do not require properties to be public here so I removed the set; from all properties.